### PR TITLE
Compass Calibrator cleanups - enumeration for status

### PR DESCRIPTION
**Pull request by peterbarker**
Bizarrely, this saves 300 bytes on the MatekF405-Wing.  It's the enum-class change saving bytes.
 
Mag cal still runs on a PixRacer here, so...

### CODE REVIEW

File: libraries/AP_Compass/CompassCalibrator.h [hunk 1]
```diff
@@ -45,8 +32,20 @@ class CompassCalibrator {
     // running is true if actively calculating offsets, diagonals or offdiagonals
     bool running() const;
 
+    // compass calibration states
+    enum class Status {
+        NOT_STARTED = 0,
```
Comment (WickedShell) [inline libraries/AP_Compass/CompassCalibrator.h]: Pre exsisting, but this should really just use the MAVLink enum values as what its assigned to rather then hand writing them in.

File: libraries/AP_Compass/CompassCalibrator.h [hunk 2]
```diff
@@ -164,7 +164,7 @@ class CompassCalibrator {
     bool fix_radius();
 
     uint8_t _compass_idx;                   // index of the compass providing data
-    enum compass_cal_status_t _status;      // current state of calibrator
+    Status _status;      // current state of calibrator
```
Comment (rmackay9) [inline libraries/AP_Compass/CompassCalibrator.h]: minor formatting issue here..

**Comment by rmackay9** [comment]
Looks pretty OK to me.  I guess it's been tested including making sure the outputs are similar before and after...

**Comment by peterbarker** [comment]
@rmackay9 Tested on PixRacer here and I got similar results before /after.

File: libraries/AP_Compass/CompassCalibrator.h (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Compass/CompassCalibrator.h]: Fixed.

File: libraries/AP_Compass/CompassCalibrator.h (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Compass/CompassCalibrator.h]: Changed to use the enumeration.

Comment (peterbarker) [reply libraries/AP_Compass/CompassCalibrator.h]: .... and then reverted, as I decided pulling in the mavlink headers to this file wasn't worth it.

**Pull request merged**